airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -42,7 +42,7 @@
 
 from airflow.exceptions import AirflowException, AirflowProviderDeprecationWarning
 from airflow.providers.cncf.kubernetes.callbacks import ExecutionMode, KubernetesPodOperatorCallback
-from airflow.providers.cncf.kubernetes.pod_generator import PodDefaults
+from airflow.providers.cncf.kubernetes.utils.xcom_sidecar import PodDefaults
```

Comment:
This is the actual fix. Use the same PodDefaults that are used when constructing the sidecar, not the version in the old deprecated pod_generator.

Reply:
These defaults are specific to Xcom sidecar. Does every sidecar has its own defaults?

Reply:
No, these are deployment wide defaults. A little more historical context - this is one area where we've instructed users to modify defaults in the past, before the ability to control the image for the sidecar directly in other ways existed (like in the connect).